ports/minimal: Make this work for macOS

yamt opened the pull request.

yamt committed:
minimal: Allow compiling on macOS.

Signed-off-by: YAMAMOTO Takashi <yamamoto@midokura.com>

dpgeorge commented on ports/minimal/Makefile:
```diff
@@ -26,9 +26,15 @@ CFLAGS_CORTEX_M4 = -mthumb -mtune=cortex-m4 -mcpu=cortex-m4 -msoft-float -fsingl
+UNAME := $(shell uname)
 LD = $(CC)
 CFLAGS += $(INC) -Wall -Werror -Wdouble-promotion -Wfloat-conversion -std=c99 $(COPT)
+ifeq ($(UNAME), Linux)
```
Please remove the space before "Linux".

yamt replied:
thank you. fixed.

i also made it similar to ports/unix and fixed the test.

dpgeorge commented on ports/minimal/Makefile:
```diff
@@ -26,9 +26,15 @@ CFLAGS_CORTEX_M4 = -mthumb -mtune=cortex-m4 -mcpu=cortex-m4 -msoft-float -fsingl
 LDFLAGS += -Wl,-Map=$@.map,--cref -Wl,--gc-sections
+else
+# assmue UNAME=Darwin
+LDFLAGS += -Wl,-map,$@.map
```
I suggest adding `-Wl,-dead_strip` to these flags here.

yamt committed:
minimal: Use printf instead of echo -e.

macOS's echo doesn't have -e option.  printf is in POSIX and more widely
available these days.

Signed-off-by: YAMAMOTO Takashi <yamamoto@midokura.com>

dpgeorge commented:
Thanks for the patch!
